metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -551,23 +611,51 @@ def _process_schema(
             yield from self.gen_schema_containers(snowflake_schema, db_name)
 
         if self.config.include_tables:
-            snowflake_schema.tables = self.get_tables_for_schema(
-                conn, schema_name, db_name
-            )
+            try:
+                snowflake_schema.tables = self.get_tables_for_schema(
+                    conn, schema_name, db_name
+                )
+            except SnowflakePermissionError as e:
+                # Ideal implementation would use PEP 678 – Enriching Exceptions with Notes
+                error_msg = f"Failed to list tables for schema {db_name}.{schema_name}. Please check permissions."
+                raise SnowflakePermissionError(error_msg) from e.__cause__
+            except Exception as e:
+                self.report_warning(
+                    "tables-for-schema",
+                    f"Failed to get tables for schema {db_name}.{schema_name} due to error {e}",
+                )
 
             if self.config.include_technical_schema:
                 for table in snowflake_schema.tables:
                     yield from self._process_table(conn, table, schema_name, db_name)
 
         if self.config.include_views:
-            snowflake_schema.views = self.get_views_for_schema(
-                conn, schema_name, db_name
-            )
+            try:
+                snowflake_schema.views = self.get_views_for_schema(
+                    conn, schema_name, db_name
+                )
+
+            except SnowflakePermissionError as e:
+                # Ideal implementation would use PEP 678 – Enriching Exceptions with Notes
+                error_msg = f"Failed to list views for schema {db_name}.{schema_name}. Please check permissions."
+
+                raise SnowflakePermissionError(error_msg) from e.__cause__
+            except Exception as e:
```

Comment:
So if we ever fail to list views or tables in a schema, the connector breaks? Is that correct

Reply:
Right. That's the intent.

If this point is reached, user already has usage permissions for database and schema. So list queries on tables and views should never fail with permission error but return an empty list of tables or views instead.

Reply:
yup